<h1 style="color: #001a79;">Lady Tasting Tea Problem:</h1>
<h2 style="color: #001a79;">Exercise 1.1</h2>

<hr style="border-top: 1px solid #001a79;" />

<h4> Calculate the minimum number of cups of tea required to ensure the probability of randomly selecting<br>
    the correct cups is less than or equal to 1%. <h4>

<b>Pre-Exercise:<b>

In [1]:
# Math module
import math

The following was adapted from https://stackoverflow.com/a/4941932.

In [2]:
# Original Problem:
# Number of ways of selecting 4 cups from 8 (i.e: 8 choose 4)
num_combinations = math.comb(8, 4)
num_combinations

70

In [3]:
# Probability of randomly selecting 4 correct cups when we dont take 
# order into consideration
probability = 1.0 / num_combinations
round(probability, 5)

0.01429


<br>

<b>Main Exercise:<b>

As shown above, when the experiment is conducted with 8 cups total (4 with milk first and 4 with tea first) there is roughly 1.4%<br>
chance of randomly selecting the 8 cups correctly. In the interest lowering this probablity to 1% lets adds 1 more cup into the mix.

In [4]:
# Number of ways of selecting 4 cups from 9
num_combinations = math.comb(9, 4)

# Probability of randomly selecting correct cups is:
probability = 1.0 / num_combinations

print("Number of combinations:  {}".format(num_combinations))
print("Probability: {}".format(round(probability, 5)))

Number of combinations:  126
Probability: 0.00794



<br>

Now we see the probability at roughly 0.79%. The reduced probability can be attributed to the higher number of combinations<br>
available for selection in the experiment, going from 70 (8 total cups) to 126 (9 total cups).

In [5]:
# Number of ways of selecting 4 cups from 10
num_combinations = math.comb(10, 4)

# Probability of randomly selecting correct cups is:
probability = 1.0 / num_combinations

print("Number of combinations:  {}".format(num_combinations))
print("Probability: {}".format(round(probability, 5)))

Number of combinations:  210
Probability: 0.00476



<br>

**Answer:**<br>
From the results above we can conclude that a total of 10 cups deviates the probability of selecting the correct cups further<br>
away from 1% target. An experiment with a total of 9 cups best alligns the probabilty of randomly selecting the correct cups<br>
based on the exercise criteria.


 <br>
 
 <h4> Bonus: How many would be required if you were to let the taster get one cup wrong while maintaining the 1% threshold? <h4>